infra-in-containers/john.md

```diff
@@ -0,0 +1,272 @@
+## From VM to Container
+### Differences between a VM and a Container
+| VM | Container |
+| ---- | ---- |
+| Memory is allocated on setup | Memory is utilized on demand |
+| Isolation is at hardware level (more secure) | Isolation is at virtual memory level (less secure) |
+| Has individual OS libraries and kernel | Shares Host's OS libraries and kernel |
+| Slower to run due to hardware emulation | Faster to run because they run directly on the host server |
+| Does not have direct access to host's hardware | Has direct access to host's hardware |
+| OS updates are done on each VM | OS updates are done on host |
+
+### Pros and cons of using a Container vs a VM
+Pros:  
+1. Light weight (usually tens of MB)
+1. Less resource intensive (no redundant OS services)
+1. Faster startup
+1. On-demand resource usage
+
+Cons:  
+1. Less secure due to isolation at a higher level
```

Comment:
[Kata Containers](https://katacontainers.io/) solves this problem by wrapping the container in a VM